### PR TITLE
Add db an media backups

### LINKED ISSUE #3778: [FR] Add backup into inventrees lifecycle
### Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### Problem statement

We are currently leaving database and media backups up to the users. As the user base seems to be less and less technical it might be good to include some basic backup into the product.

### Suggested solution

I use [dbbackup](https://github.com/jazzband/django-dbbackup) in other codebases, it works reasonable enough and would be compatible with the storage backend - should we choose to support k8 in the future.

### Describe alternatives you've considered

The current way works for seasoned sysadmins.

### Examples of other systems

Snipe-IT, Nextcloud have good backup systems.

### Do you want to develop this?

- [X] I want to develop this.

### PR DESCRIPTION
**Pull request by matmair**
Missing:
- [x] Update Package installer(#3743)  for this 
- [x] Regular backup task
- [x] Adjust docker
- [x] Docs

Closes #3778

<a href="https://gitpod.io/#https://github.com/inventree/InvenTree/pull/3783"></a>

### CODE REVIEW

File: InvenTree/InvenTree/config.py [hunk 1]
```diff
@@ -149,6 +149,22 @@ def get_static_dir(create=True):
     return sd
 
 
+def get_backup_dir(create=True):
+    """Return the absolute path for the backup directory"""
+
+    bd = get_setting('INVENTREE_BACKUP_ROOT', 'backup_root')
```
Comment (SchrodingersGat) [inline InvenTree/InvenTree/config.py:155]: We should use `BACKUP_DIR` not `BACKUP_ROOT` here

File: tasks.py [hunk 2]
```diff
@@ -196,7 +196,27 @@ def translate(c):
...
+def restore(c):
+    """Restore the database and media files."""
+
+    print("Restoring InvenTree database...")
+    manage(c, "dbrestore --noinput --uncompress")
+    print("Restoring InvenTree media files...")
+    manage(c, "mediarestore --noinput --uncompress")
+
+
+@task(pre=[backup, ], post=[rebuild_models, rebuild_thumbnails])
```
Comment (SchrodingersGat) [inline tasks.py:219]: Nice, I like the addition of automated pre-update backup

**Review by SchrodingersGat — COMMENTED** [review]
We also need to adjust the dockerfile with a default path to the backup directory.

**Comment by matmair** [comment]
@SchrodingersGat this should be all done now;
The current setup limits the backup count to 10 and then deletes the oldest after making a successful run. In a further PR I would like to add a friendly interface to the admin site and maybe system checks later.

Gone stop working on this for now as feature creep tends to make my PRs huge and diverge my fork (to far) from mainline 😇.

File: tasks.py [hunk 3]
```diff
@@ -196,7 +196,27 @@ def translate(c):
...
+    """Backup the database and media files."""
+
+    print("Backing up InvenTree database...")
+    manage(c, "dbbackup --noinput --clean --compress")
+    print("Backing up InvenTree media files...")
+    manage(c, "mediabackup --noinput --clean --compress")
+
+
+@task
+def restore(c):
```
Comment (SchrodingersGat) [inline tasks.py:210]: Can we allow the user to specify *which* backup to restore from?

File: InvenTree/InvenTree/apps.py [hunk 4]
```diff
@@ -117,6 +117,12 @@ def start_background_tasks(self):
             schedule_type=Schedule.DAILY
         )
 
+        # Make regular backups
+        InvenTree.tasks.schedule_task(
```
Comment (SchrodingersGat) [inline InvenTree/InvenTree/apps.py:121]: Can we give the user (or more likely, sysadmin) a method to customize:

a) Whether backups are performed automatically or not
b) The schedule period for performing backups

File: tasks.py (hunk 3 above)
Comment (matmair) [reply tasks.py:210]: Yes, that is theoretically possible; Current method is to use the latest

File: InvenTree/InvenTree/apps.py (hunk 4 above)
Comment (matmair) [reply InvenTree/InvenTree/apps.py:121]: a) Yes, easy
b) Not currently implemented in the registering logic so I'd rather not put this in here and wait if there are user requests for it

**Review by SchrodingersGat — APPROVED** [review]

**Comment by SchrodingersGat** [comment]
https://github.com/inventree/InvenTree/actions/runs/3259581495/jobs/5352510594

@matmair we'll need to add some more system package requirements to various installer methods based on this

**Pull request merged**